src/circuit/gadget/poseidon/pow5.rs

```diff
@@ -781,14 +785,15 @@ mod tests {
                         Ok(CellValue::new(cell, value))
                     };
 
-                    Ok([message_word(0)?, message_word(1)?])
+                    let message: Result<Vec<_>, Error> = (0..RATE).map(message_word).collect();
```

Comment:
This should be iterating over the constant length being hashed, not the permutation rate:
```suggestion
                    let message: Result<Vec<_>, Error> = (0..L).map(message_word).collect();
```
(The current change would cause a panic if `RATE > 2`, because `message_vals[i]` above would index out of bounds.)